web/src/components/betaBar/BetaBar.test.tsx

```diff
@@ -0,0 +1,178 @@
+      });
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeedbackButtonText)).toBeInTheDocument();
+      expect(screen.getByTestId("feedback-link").getAttribute("href")).toEqual(
+        Config.betaBar.betaFormLinkOwning
+      );
+    });
+
+    it("links to correct feedback form when completed onboarding and starting a business", () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeedbackButtonText)).toBeInTheDocument();
+      expect(screen.getByTestId("feedback-link").getAttribute("href")).toEqual(
+        Config.betaBar.betaFormLinkStarting
+      );
+    });
+  });
+
+  describe("request a feature", () => {
+    it("does not show feature request button when userData is undefined", () => {
+      useUndefinedUserData();
+      render(<BetaBar />);
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.queryByText(Config.betaBar.betaModalFeatureRequestButtonText)).not.toBeInTheDocument();
+    });
+
+    it("clears feature request input when modal is closed and is revisted", () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      fireEvent.change(screen.getByLabelText("Feature request"), { target: { value: "random text" } });
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("random text");
+
+      fireEvent.click(screen.getByLabelText("close"));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText));
+      expect((screen.getByLabelText("Feature request") as HTMLInputElement).value).toBe("");
+      expect(screen.queryByText(Config.betaBar.featureRequestModalInlineErrorText)).not.toBeInTheDocument();
+    });
+
+    it("submitting feature request renders submission successful modal", async () => {
+      useMockUserData({
+        formProgress: "COMPLETED",
+        profileData: generateProfileData({ hasExistingBusiness: false }),
+      });
+
+      render(<BetaBar />);
+
+      fireEvent.click(screen.getByText(Config.betaBar.betaModalButtonText));
+      expect(screen.getByText(Config.betaBar.betaModalFeatureRequestButtonText)).toBeInTheDocument();
```

Comment:
once we've have a test for something, I think we don't need to re-test and re-expect it in every test. If we have a test for "it only shows the Feature Request button when form progress is completed" then we don't need this line in every subsequent test. This helps with readability by keeping our tests concise so that every line is necessary and nothing is extraneous.

Reply:
This holds for every test